R/agg_scale.R

```diff
@@ -57,6 +55,53 @@
 #' @return \[`data.table()`\] with `id_cols` and `value_cols` columns for
 #'   requested aggregates or with scaled values.
 #'
+#' @section Severity Arguments:
+#' **`missing_dt_severity`**:
+#'
+#' Check for missing levels of `col_stem`, the variable being aggregated or
+#' scaled over.
+#' 1. `stop`: throw error (this is the default).
+#' 2. `warning` or `message`: throw warning/message and continue with
+#' aggregation/scaling where possible.
+#' 3. `none`: don't throw error or warning, continue with aggregation/scaling
+#' where possible.
+#' 4. `skip`: skip this check and continue with aggregation/scaling.
+#'
+#' **`present_agg_severity`** (`agg` only):
+#'
+#' Check for requested aggregates in `mapping` that are already present
+#' 1. `stop`: throw error (this is the default).
+#' 2. `warning` or `message`: throw warning/message, drop aggregates and continue
+#' with aggregation.
+#' 3. `none`: don't throw error or warning, drop aggregates and continue with
+#' aggregation.
+#' 4. `skip`: skip this check and add to the values already present for the
```

Comment:
What about skipping aggregation for groups w/ aggregates already present? So for groups w/o aggregates we'd want to add aggregates and for groups w/ aggregates we'd just take the original aggregate values?

Reply:
How difficult is it to have both options? If only one option, I'd say default to throw warning/message and then skip aggregation for existing aggregates. It's easier for people to set their code to drop existing aggregates before aggregation than to remove existing aggregates and append them back on in such a way that they replace new aggregates.

Reply:
Created a separate issue for this https://github.com/ihmeuw-demographics/hierarchyUtils/issues/57